### PR TITLE
Ensure storage pool tests use an empty directory

### PR DESCRIPTION
**Pull request by jgraichen**
```
All tests involving a storage pool used `/mnt` has the hard-coded source
for the pool. This fails if the hosts `/mnt` is *not* empty. This might
actually not be true and in some cases (e.g. Github Actions VM) it
cannot be changed at all by the developer.

This CL changes all tests to use a new temporary directory, which
*always* is empty as the pool source.

The Go 1.15 update is needed as with 1.15 the testing frameworks
provides a ready to use function to get a temporary directory in a test
case.
```

I've hit this issue locally a few times when having a mount in `/mnt` and on Github Actions. I did include the Go 1.15 upgrade here because it provide such a simple and elegant solution. I've fixed a few wrong indentions within HCL codeblocks too.

### CODE REVIEW

**Comment by jgraichen** [comment]
Would you accept a PR that adds Github Actions (GhA) for running the tests for stable Go and LXD? Travis CI is becoming extremely slow on travis-ci.org and I am still waiting for new open source free credits to run on travis-ci.com from my fork.

Unfortunately GhA does not support allowed failures making automated tests for LXD edge and Go master not feasible.

**Comment by jtopjian** [comment]
@jgraichen I think moving to Github Actions is a good idea. I haven't looked at the Travis changes in detail, but from what I read, it's sounding like it's best long-term to move away from it.

As far as the "allowed failures", I think we can remove those.

@sl1pm4t and I are trying to get some time to move this repository to a centralized Github org. Once that is done, we can look at PRs to set up Github Actions.

**Comment by jgraichen** [comment]
@jtopjian Thanks for the update! Out of curiosity, are you thinking about creating a new org or looking for an existing one such as [github.com/lxc](https://github.com/lxc)?

**Comment by jtopjian** [comment]
@jgraichen a new org (as has just recently happened :slightly_smiling_face:)

Please feel free to add any Github Actions to the `.github` directory. Let me know if any of them need special permissions.

**Comment by jgraichen** [comment]
Thanks!

Shall I add the [workflow](https://github.com/jgraichen/terraform-provider-lxd/blob/m/github-actions/.github/workflows/test.yml) here, as it depends on this change, or in a separate, additional PR?

**Comment by jtopjian** [comment]
@jgraichen Sure - let's give it a shot. Adding it to this PR is fine.

**Comment by jgraichen** [comment]
Rebased and updated initial commit. Added test workflow in second commit, and it is running.

**Comment by jtopjian** [comment]
@jgraichen Nice - looks like it's working. It's interesting that any PR can trigger a new action, though...

Are you finished working on the storage pool tests? If so, I can merge this.

**Comment by jgraichen** [comment]
@jtopjian Actions run on PRs have limited access within Github Actions as far as I know. They should not be able access configured secrets etc. Everything else should be free for public repositories.

This PR is ready to be merged. I've only updated the tests to not assume `/mnt` is in empty directory, which also does not apply on Github Actions. As all tests as passing it is working as expected.

**Comment by jgraichen** [comment]
I will rebase #215 as soon as this is merged to have to run tests on GhA too.

**Review by jtopjian — APPROVED** [review]
LGTM - thank you!

**Pull request merged**
